modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -286,6 +291,15 @@ func (p *Processor) onComplete(e *store.Edge) {
 
 	p.serviceGraphRequestServerSecondsHistogram.ObserveWithExemplar(registryLabelValues, e.ServerLatencySec, e.TraceID, e.SpanMultiplier)
 	p.serviceGraphRequestClientSecondsHistogram.ObserveWithExemplar(registryLabelValues, e.ClientLatencySec, e.TraceID, e.SpanMultiplier)
+
+	if p.Cfg.EnableMessagingSystemLatencyHistogram && e.ConnectionType == store.MessagingSystem {
```

Comment:
this is looking really good. i think my final question is just related to this recent discussion item:

https://github.com/grafana/tempo/discussions/3616

Is there any reason in particular to limit this change to producer/consumer spans? Does making this apply to client/server as well help the request above?

cc @jSwanson99

Reply:
The calculation I did here won't help with what @jSwanson99 described in his [screenshot](https://github.com/grafana/tempo/discussions/3616#discussion-6565351).
I'm looking at the diff between the producer ("client") start time and the consumer ("server") start time.

So anyway, I don't see it relevant for this PR...
Anyway - commented there my opinion 🙃

Reply:
@joe-elliott let me know if we're clear to rebase again and merge 🥳

Reply:
sorry friend. we have worked on this long enough. after you catch up to main I will approve and merge. it will be easy enough to extend if @jSwanson99 feels like it will solve his use case. 

apologies for the delay.

Reply:
No worries,
I will sync to main tomorrow morning and let you know 🙂